compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -2666,6 +2687,10 @@ impl<'tcx> ty::Binder<'tcx, ty::TraitRef<'tcx>> {
     pub fn print_only_trait_path(self) -> ty::Binder<'tcx, TraitRefPrintOnlyTraitPath<'tcx>> {
         self.map_bound(|tr| tr.print_only_trait_path())
     }
+
+    pub fn print_trait_sugared(self) -> ty::Binder<'tcx, TraitRefPrintSugared<'tcx>> {
+        self.map_bound(|tr| tr.print_trait_sugared())
+    }
```

Comment:
Should we not just have made this part of `print_only_trait_path`?

Reply:
`print_only_trait_path` is used in other places where the suggestion would be misleading or syntactically incorrect. Basically any place where the printed trait ref would be used in a structured suggestion.